### ISSUE TITLE
extremely long next build crashes with `JavaScript heap out of memory`

### ISSUE THREAD

**Issue by omerdn1** [zeit/next.js#3391]
So we've been trying to build our project for production and we ran into this build issue.
The build takes around 45 minutes to an hour and after that it crashes with: `FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory`

Our `next.config.js`:
```const path = require('path');
const glob = require('glob');
const Dotenv = require('dotenv-webpack');

module.exports = {
  webpack: (config, { dev }) => {
    config.module.rules.push(
      {
        test: /\.(css|scss)/,
        loader: 'emit-file-loader',
        options: {
          name: 'dist/[path][name].[ext]',
        },
      },
      {
        test: /\.css$/,
        use: ['babel-loader', 'raw-loader', 'postcss-loader'],
      },
      {
        test: /\.s(a|c)ss$/,
        use: [
          'babel-loader',
          'raw-loader',
          'postcss-loader',
          {
            loader: 'sass-loader',
            options: {
              includePaths: ['styles', 'node_modules']
                .map(d => path.join(__dirname, d))
                .map(g => glob.sync(g))
                .reduce((a, c) => a.concat(c), []),
            },
          },
        ],
      }
    );
    config.plugins.push(new Dotenv())
    return config;
  }
};
```

and our `postcss.config.js`:
```/* eslint-disable global-require */
module.exports = {
  plugins: [
    require('postcss-easy-import')({ prefix: '_' }), // keep this first
    require('autoprefixer')({ /* ...options */ }), // so imports are auto-prefixed too
  ],
};
```
After reading online some people suggested to remove `postcss-easy-import` but it didn't seem to do the job.

## Environment

| Tech    | Version |
|---------|---------|
| next    |   4.1.4      |

**Comment by delta98** [comment]
How many pages are you building?

**Comment by bentatum** [comment]
i'm having a similar issue. this seems related.

https://github.com/webpack/webpack/issues/1914

**Comment by timneutkens** [comment]
Fixed in 4.2.1
